src/openforms/registrations/contrib/objects_api/config.py

```diff
@@ -92,3 +136,41 @@ class ObjectsAPIOptionsSerializer(JsonSchemaSerializerMixin, serializers.Seriali
         ],
         required=False,
     )
+
+    # V2 only fields:
+    variables_mapping = ObjecttypeVariableMappingSerializer(
+        label=_("variables mapping"),
+        many=True,
+        required=False,
+    )
+
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        v1_only_fields = {"content_json", "payment_status_update_json"}
+        v2_only_fields = {"variables_mapping"}
+
+        version = attrs["version"]
```

Comment:
technically this serializer could be called via a PATCH call where this key is missing from the `attrs`. You can use `openforms.api.get_from_serializer_data_or_instance` to address this.